test/responses_unit_test.go

```diff
@@ -0,0 +1,222 @@
+package test
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/cucumber/godog"
+	"github.com/nsf/jsondiff"
+
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+	"github.com/algorand/go-algorand-sdk/client/v2/indexer"
+	"github.com/algorand/go-algorand-sdk/encoding/json"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+var algodC *algod.Client
+var indexerC *indexer.Client
+var baselinePath string
+var expectedStatus int
+var response interface{}
+
+// @unit
+// @unit.responses
+
+func mockHttpResponsesInLoadedFromWithStatus(
+	jsonfile, loadedFrom /* generated_responses*/ string, status int) error {
+	directory := path.Join("./features/resources/", loadedFrom)
+	baselinePath = path.Join(directory, jsonfile)
+	var err error
+	expectedStatus = status
+	err = mockHttpResponsesInLoadedFromHelper(jsonfile, directory, status)
+	if err != nil {
+		return err
+	}
+	algodC, err = algod.MakeClient(mockServer.URL, "")
+	if err != nil {
+		return err
+	}
+	indexerC, err = indexer.MakeClient(mockServer.URL, "")
+	return err
+}
+
+func weMakeAnyCallTo(client /* algod/indexer */, endpoint string) (err error) {
+	var round uint64
+	var something interface{}
+
+	switch client {
+	case "indexer":
+		switch endpoint {
+		case "lookupAccountByID":
+			round, something, err = indexerC.LookupAccountByID("").Do(context.Background())
+			response = models.LookupAccountByIDResponse{
+				CurrentRound: round,
+				Account:      something.(models.Account),
+			}
+		case "searchForAccounts":
+			response, err = indexerC.SearchAccounts().Do(context.Background())
+		case "lookupApplicationByID":
+			response, err = indexerC.LookupApplicationByID(10).Do(context.Background())
+		case "searchForApplications":
+			response, err = indexerC.SearchForApplications().Do(context.Background())
+		case "lookupAssetBalances":
+			response, err = indexerC.LookupAssetBalances(10).Do(context.Background())
+		case "lookupAssetByID":
+			round, something, err = indexerC.LookupAssetByID(10).Do(context.Background())
```

Comment:
I see that this is still returning 3 things instead of a response object and an error?

`SearchForAssets` was changed to return a response object. Why are only some of these `Do` functions updated?

Reply:
I am not changing the existing API. 
The only published one changed (AssetsResponse) is because it has a bug anyway.
This is not a bug, just not consistent. And changing it will break the published API. 
If we are to change published API, I'll replace the whole v2 with the generated one :-)